### DISCUSSION TITLE
Working API Endpoints for Transactions???

### DISCUSSION THREAD

**Discussion by bigshooTer39** [cosmos/gaia#2315] [Q&A]
Is there an endpoint that is open for public use to pull transactions by account for each Cosmos SDK chain?   Maybe a different base url for each chain?

All I want to do is download my transactions for the chains in the Cosmos Ecosystem.  If mintscan enabled csv export or offered a public endpoint, I would be all set.  For some reason, not a single explorer in Cosmos offers this functionality. 

For all of the endpoints I am able to find, it seems that "transactions by account" are deprecated.  Why is this?  For example, I am able to pull details of a transaction by txHash for Kava, but I can't pull the list of tx hashes to loop through that endpoint.

Can anyone help?  Looking to pull all my trx for atom, juno, evmos, scrt, osmo, kava, inj, jkl, rebus, etc....

As of right now, I am 100% reliant on stake.tax for exporting 60% of Cosmos portfolio.  The other 40%, I have to key in each transaction 1 by 1.  It is painful and extremely time consuming.

I was able to find working a working endpoint for Crypto.Org chain and have that all sorted out.  I used [https://crypto.org/explorer/api/v1/account/[enter address]].  Unfortunately that is the only chain I am able to get working.

I am frustrated to the point that I started teaching myself Python in an attempt to access my transactional history in Cosmos.   Struggling but learning.

**Comment by mmulji-ic** [comment]
Hi @bigshooTer39 you can use the `gaiad` application and the public rpc endpoints in the [chain-repository](https://github.com/cosmos/chain-registry/blob/master/cosmoshub/chain.json) to download your transactions. You should be able to automate this using a small shell script. Below I use `polkachu's` public rpc endpoint.

`gaiad query txs --events 'message.sender=...' --node https://cosmos-rpc.polkachu.com:443`
Remember to substitute the ... with your account number.

You can also use `--output json` to output json format data which you can parse using the [jq](https://stedolan.github.io/jq/) tool.

For further documentation and examples take a look at the [Gaia cli docs](https://hub.cosmos.network/main/hub-tutorials/gaiad.html#query-transactions).

**Reply by bigshooTer39** [reply]
Thank you for this.  I'm familiar with the chain repository.  I use it all the time for nodes/endpoints.  I haven't checked my notifications in a while, which is why i just replying.  

Does message.sender give you everything?   Maybe this is a dumb question but why am I returning trx where my address is the receiver using the `message.sender` parameter.   Is it because I initiated the trx, making me the message sender?  I was expecting to need to also use `message.receiver` however that doesn't exist under "txs"

 `"base url/cosmos/tx/v1beta1/txs?events=message.sender=%27myaddress%27"`. 

If I just wanted to pull auto-compound transactions (restake), which parameter would I use?  I can't seem to figure this out.  Looking at the attributes under tx_responses, I think I would want something like this but no luck.  I gave up trying to get it to work.

`https://api.data.kava.io/cosmos/tx/v1beta1/tx_responses?events=logs.Events&events.type='coin_received'&attributes.receiver='myaddress'`


I've been using PowerQuery bc I'm a novice with CLI.   I do have MS visual studio installed but the only repository I have installed is stake.tax.  It obviously errors when I run that script you shared.  I'll need to get setup with the repository from the chain registry.
